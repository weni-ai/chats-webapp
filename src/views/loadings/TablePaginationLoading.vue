<template>
  <section class="table-pagination-loading">
    <unnnic-skeleton-loading class="table-pagination-loading__count" />
    <div class="table-pagination-loading__pagination">
      <unnnic-skeleton-loading v-for="page in 7" :key="`page-${page}`" height="36px" width="36px" />
    </div>
  </section>
</template>

<script>
export default {
  name: 'TablePaginationLoading',
};
</script>

<style lang="scss" scoped>
.table-pagination-loading {
  margin-top: $unnnic-spacing-md;
  border-top: $unnnic-border-width-thinner solid $unnnic-color-neutral-soft;
  padding-top: $unnnic-spacing-md;

  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: $unnnic-spacing-sm;

  &__count {
    height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
    width: 84px;
  }

  &__pagination {
    display: flex;
    gap: $unnnic-spacing-xs;
  }
}
</style>
