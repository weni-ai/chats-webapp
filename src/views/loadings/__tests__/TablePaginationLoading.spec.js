import { mount } from '@vue/test-utils';
import TablePaginationLoading from '../TablePaginationLoading.vue';
import { beforeEach, describe, expect } from 'vitest';

const createWrapper = () => {
  return mount(TablePaginationLoading);
};

describe('TablePaginationLoading.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
