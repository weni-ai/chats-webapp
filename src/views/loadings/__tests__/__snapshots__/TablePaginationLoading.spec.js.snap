// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TablePaginationLoading.vue > to match snapshot 1`] = `
"<section data-v-6dcaf26f="" class="table-pagination-loading"><span data-v-6dcaf26f="" class="table-pagination-loading__count"><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span>
  <div data-v-6dcaf26f="" class="table-pagination-loading__pagination"><span data-v-6dcaf26f=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 36px; height: 36px;"> ‌ </span></span><span data-v-6dcaf26f=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 36px; height: 36px;"> ‌ </span></span><span data-v-6dcaf26f=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 36px; height: 36px;"> ‌ </span></span><span data-v-6dcaf26f=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 36px; height: 36px;"> ‌ </span></span><span data-v-6dcaf26f=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 36px; height: 36px;"> ‌ </span></span><span data-v-6dcaf26f=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 36px; height: 36px;"> ‌ </span></span><span data-v-6dcaf26f=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; width: 36px; height: 36px;"> ‌ </span></span></div>
</section>"
`;
